The logo can be changed if preferred

Koshila-cd opened the pull request:
A logo parameter is added

dinukadesilva commented:
@Koshila-cd It's great work and glad about your intrest in contributing to app-header.

Please upload a screenshot of the new addition.

Also, it would be nice if you could rebase and update the author of all the commits. So, that all commits will be linked to your profile.

CC: @wyseguyonline @mhomolak

dinukadesilva commented on src/js/template.js:
```diff
@@ -17,17 +17,18 @@ function template (data, handlers, translate) {
             elementOpen("div", null, ["class", "o-header__logo o-header__logo--pearson"])
-              elementOpen("img", null, null, "src", images.logoLarge, "alt", translate('Pearson - Home'))
+            elementOpen("img", null, null, "src", logo, "alt", translate('Pearson - Home'))
               elementClose("img")
```
I think you should not change this file manually and it has to be generated from `src/html/header.template`. Or have you missed to add `src/html/header.template` file to commit?
![image](https://user-images.githubusercontent.com/10680296/41348827-7f023b82-6f2b-11e8-94a5-e55a430e2594.png)

Koshila-cd replied:
@dinukadesilva Thank you very much for the assistance. No it's not that I didn't commit the `header.template` file. I will change it accordingly and generate the code.

dinukadesilva commented on README.md:
```diff
@@ -136,13 +136,18 @@ Creates a new instance of `AppHeader`.
 | links                    | `object`               | Links in the user dropdown. Refer to the [Links](#links) section for the default list  |
 | mode                     | `string`               | The mode. Refer to the [Modes](#modes) section for a list of supported modes and options. |
-
+| logo                     | `string`               | Logo in the left side of the header (default: Pearson logo) |
```
If you can add an example too. That's great.

Koshila-cd replied:
@dinukadesilva I actually have added a description

![image](https://user-images.githubusercontent.com/35322940/41393260-8841d58a-6fc2-11e8-8c90-a4cb66bc5b32.png)

And comment if it would not be enough.

Koshila-cd commented:
Thanks for the assistance @dinukadesilva , and attached here is a screenshot of the logo change of the app-header.

![image](https://user-images.githubusercontent.com/35322940/41347849-89d96614-6f28-11e8-94bf-45934a34d409.png)

@antelopeb @mhomolak

Koshila-cd commented:
@dinukadesilva @antelopeb 
Can you verify what is wrong in my files for the test cases to fail? Or is it because i'm in a fork branch.

Thanks in Advance.

antelopeb commented:
Yeah, it's because the environment variables aren't present, so it can't build properly. Will take a look now!

antelopeb commented:
@Koshila-cd -- I believe the design requirement was to allow for a text string after the Pearson logo. Please see the designs from the UX team:
![image](https://user-images.githubusercontent.com/1117763/41479883-e8a49af8-7089-11e8-97e6-fee7a86983f1.png)

dinukadesilva commented:
Also somethig to note is @antelopeb, There are some other teams having following style as well.
![image](https://user-images.githubusercontent.com/10680296/40907357-2eef8c98-6801-11e8-827b-13480891d1f7.png)


Either directing all the teams come to a one design could be an option, Or, what if both icons are considered as two customization icons. Company Logo and Application Logo.

Koshila-cd commented:
@antelopeb  @dinukadesilva 

Here's the new change done to the app-header, the company logo and the application logo can be passed seperately.
![image](https://user-images.githubusercontent.com/35322940/41543236-325d9cec-7334-11e8-84d3-3327ae411901.png)


Here I have given an example by adding the company logo.
![image](https://user-images.githubusercontent.com/35322940/41543186-1d75a0d6-7334-11e8-9ed2-ad8df456488a.png)


Here I have added a screenshot of the requirement of my team "Statcrunch"
![image](https://user-images.githubusercontent.com/35322940/41543476-7f7770b6-7334-11e8-8c3f-d008dcfb1bde.png)

I have added an image input to the application logo since our team the application logo as it is.

Please clarify or suggest if any change is required.

antelopeb commented:
@Koshila-cd -- does it still allow for a default behavior, where a team passes in nothing and it appears as it did before?

Sorry for the delay in reviewing. I will be trying to do this in the next few days.

Koshila-cd commented:
@antelopeb 
No worries!

Yes the default logo is shown when the user doesn't pass any logo and let me show some screenshots: 

**1) When user doesn't pass any logo**

![image](https://user-images.githubusercontent.com/35322940/42199338-950f4502-7eab-11e8-916c-fa7117926b41.png)


**2) When user passes one logo (Most probably the short Pearson logo)**

![image](https://user-images.githubusercontent.com/35322940/42199615-06212156-7ead-11e8-98a8-925c78f58b86.png)

**3) When user passes both logos (Company logo and application logo)**

![image](https://user-images.githubusercontent.com/35322940/42199646-3846188a-7ead-11e8-8c2d-09ec816b05b5.png)


The above logos are images.

**There are some more changes to be done, you can start reviewing when I have completed doing changes. (By end of today (Sri Lankan Time)).

antelopeb commented:
@Koshila-cd Can you also check for full responsiveness in your code? Sorry to keep throwing things at you, but this is looking great so far!! If you can do the responsive piece I should be able to merge quickly!!

Reference to design: https://uxframework.pearson.com/c/special-application-header/

Koshila-cd commented:
@antelopeb The Mobile responsiveness for the Pearson logo is added, i.e when the user adds an application logo, the Pearson text of the default Pearson logo is removed from Width 768px and below, and only the Small Pearson(P) icon will be remaining.

Width = 768px :
![image](https://user-images.githubusercontent.com/35322940/43816289-7f6379dc-9af1-11e8-908a-c151765bcbe8.png)

Width = 767px : 
![image](https://user-images.githubusercontent.com/35322940/43816336-bd6965f2-9af1-11e8-9992-93dd286d99c5.png)

davodey commented:
@Koshila-cd can you look at the Travis build and see if you can fix the errors?  Also, I am having issues trying to run the dev and build scripts, I get errors.  I'm not sure how long it's been like this, so please let me know if I need to disregard, but it seems like something we should have fixed.

CC @antelopeb

Koshila-cd commented:
@davodey The errors of Travis build is probably because I don't have access to the repository or I'll have a look in it. And about the dev and build errors, if you can post a screenshot then maybe I can have a look in it.
Thanks.

CC: @antelopeb

davodey commented:
@Koshila-cd it looks like the script errors are in master.  Im reaching out to others to see if its something we need to be concerned with.
